### PR TITLE
Revisit documentation

### PR DESCRIPTION
**Pull request by jingibus**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,20 +46,86 @@ dependencies {
...
+  val turbine1 = flowOf(1).testIn(backgroundScope)
+  val turbine2 = flowOf(2).testIn(backgroundScope)
+  assertEquals(1, turbine1.awaitItem())
+  assertEquals(2, turbine2.awaitItem())
+  turbine1.awaitComplete()
+  turbine2.awaitComplete()
+}
+```
+
+Like `test`, `testIn` produces a `ReceiveTurbine`.
```
Comment (JakeWharton) [inline README.md:113]: `ReceiveTurbine` as a type was not mentioned above in the `test` section.

That section probably just needs a quick tweak like
> Then it calls the validation block, passing in the read-only interface `ReceiveTurbine` as the receiver:

File: README.md [hunk 2]
```diff
@@ -127,22 +199,87 @@ Failure to consume an error will result in the same unconsumed event exception a
...
+While flow termination is exposed as an event, it is not really an event: it's a state that the flow is in.
+So completion or error events can be observed multiple times.
+
+```kotlin
+flow<Nothing> { throw RuntimeException("broken!") }.test {
+  awaitError()
+  awaitError()
+  awaitError()
+}
 ```
```
Comment (JakeWharton) [inline README.md]: Not sure whether this meets that bar for the intro/README. It's a great snippet to put on the kdoc of each of these functions (which we should do no matter what), but I'm unsure as to whether it's something important enough to actively communicate here. What do you think?

**Review by JakeWharton — APPROVED** [review]
Very nice

File: README.md (hunk 2 above)
Comment (jingibus) [reply README.md]: Makes sense to me. I was very worried about this README getting too long, so.... happy to have a reason to delete something

Comment (jingibus) [reply README.md]: This is already documented well enough, I think:

```
  /**
   * Assert that the next event received was an error terminating the flow.
   * This function will suspend if no events have been received.
   *
   * When this [ReceiveTurbine] is in a terminal state ([Event.Complete] or [Event.Error], this method
   * will yield the same result every time it is called.
   *
   * @throws AssertionError if the next event was an item or completion.
   */
```

**Pull request merged**